qp_klp/AmpliconStep.py

```diff
@@ -0,0 +1,102 @@
+from os import listdir, makedirs
+from os.path import join, isfile, basename
+from shutil import copyfile
+from qp_klp.Step import Step
+
+
+class AmpliconStep(Step):
+    def __init__(self, pipeline, master_qiita_job_id, sn_tid_map_by_project,
+                 status_update_callback=None):
+        super().__init__(pipeline,
+                         master_qiita_job_id,
+                         sn_tid_map_by_project,
+                         status_update_callback)
+
+        if pipeline.pipeline_type != 'amplicon':
+            raise ValueError("Cannot instantiate AmpliconStep object from "
+                             f"pipeline of type '{pipeline.pipeline_type}'")
+
+    def convert_bcl_to_fastq(self):
+        config = self.pipeline.configuration['bcl2fastq']
```

Comment:
Is this hardcoded? Just asking but if is maybe adding a note about it would be good.

Reply:
No less than previously in qp-klp but yeah, the knowledge to look for 'bcl2fastq' for Amplicon and 'bcl-convert' for Metagenomics is hardcoded. I can think about how to do that in a more thoughtful way, or listen to suggestions. For now I'll add a comment.